### DISCUSSION TITLE
Whatsapp Auth

### DISCUSSION THREAD

**Discussion by LeDragunov** [supabase/supabase#6717] [Feature Requests]
I suggest to have a magic link sent through WhatsApp for authentication.

There are two approaches.
Approach 1:
1) User chooses WhatsApp a method to sign-in
2) redirected to whatssap, with a message "sign-in" ready to be sent to bot
3) once sent by the user
4) the bot replys back with a magic link to sign-in

Approach 2:
1) User chooses WhatsApp a method to sign-in
2) User enters his whatsapp number
3) User recieves a magic link to sign-in

**Comment by timdiacon** [comment]
Is there any reason why you can already do this by generating a magic link serverside and sending to whatsapp via the whatsapp api?

**Reply by step135** [reply]
The reason is simple: save the time!!!

**Comment by iampopal** [comment]
Looking at this feature, It will be significant to user receive a pin code and entry verification pin code and login

**Comment by ChinKX** [comment]
Hi, the Supabase team just added a new feature for Whatsapp OTP login. I am not sure if this is what you want but I hope this will help!

![image](https://github.com/supabase/supabase/assets/40538074/d06c9a9e-e962-443c-a204-79455d4fc643)

You can visit this link for more details https://supabase.com/docs/guides/auth/phone-login/twilio
